commons-numbers-complex/src/test/java/org/apache/commons/numbers/complex/ComplexComposeTest.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.commons.numbers.complex;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+class ComplexComposeTest {
+
+    private static final ComplexUnaryOperator<ComplexDouble> neg = ComplexFunctions::negate;
+    private static final ComplexUnaryOperator<ComplexDouble> multiplyImag = ComplexFunctions::multiplyImaginary;
+    private static final ComplexUnaryOperator<ComplexDouble> conj = ComplexFunctions::conj;
+    private static final ComplexUnaryOperator<ComplexDouble> multiplyImagConj = multiplyImag.andThen(conj);
+    private static final ComplexUnaryOperator<ComplexDouble> conjMultiplyImag = conj.andThen(multiplyImag);
+    private static final ComplexUnaryOperator<ComplexDouble> identity1 = multiplyImagConj.andThen(multiplyImagConj);
+    private static final ComplexUnaryOperator<ComplexDouble> identity2 = conjMultiplyImag.andThen(conjMultiplyImag);
+
+    @Test
+    void testConjugateIdentities() {
```

Comment:
This type of test is better as a `@ParameterizedTest` using a `@CsvSource` to provide the (r, i) pairs. Since the same numbers are used for two tests you could use a `@MethodSource` for the two test methods that require a single number:
```Java

    static Stream<Arguments> singleComplex() {
        return Stream.of(
            Arguments.of(3, 4),
            Arguments.of(3, -4)
        );
    }

    @ParameterizedTest
    @MethodSource(value = {"singleComplex"})
    void testConjugateIdentity(double r, double i) {
        Complex c = Complex.ofCartesian(r, i);

```